lib/matplotlib/backends/backend_svg.py

```diff
@@ -376,14 +395,14 @@ def _write_metadata(self, metadata):
                     elif isinstance(d, (datetime.datetime, datetime.date)):
                         dates.append(d.isoformat())
                     else:
-                        raise ValueError(
-                            'Invalid type for Date metadata. '
-                            'Expected iterable of str, date, or datetime, '
-                            'not {!r}.'.format(type(d)))
+                        raise TypeError(
```

Comment:
This needs an API change note and it is changing the exception type on public API.